lib/lipseys/order.rb

```diff
@@ -24,32 +24,48 @@ class Order < Base
       ).freeze
     }
 
+    SUBMIT_TO_DROP_SHIP_FIREARM_ATTRS = {
+      permitted: %i( ffl po name phone delay_shipping disable_email items item_no quantity note ).freeze,
+      required:  %i( ffl po name phone items item_no quantity ).freeze
+    }
+
     ENDPOINTS = {
-      submit_to_store:     "order/apiorder".freeze,
-      submit_to_drop_ship: "order/dropship".freeze,
+      submit_to_store:             "order/apiorder".freeze,
+      submit_to_drop_ship:         "order/dropship".freeze,
+      submit_to_drop_ship_firearm: "order/dropshipfirearm".freeze,
     }
 
     def initialize(client)
       @client = client
     end
 
-    def submit_to_store(order_data)
+    def submit_order(order_data, arrangement)
+      # Using the SUBMIT_TO_STORE_ATTRS as a base check, since all requests should have at least that data
       requires!(order_data, *SUBMIT_TO_STORE_ATTRS[:required])
 
-      endpoint = ENDPOINTS[:submit_to_store]
+      case arrangement
+      when :transfer_to_3rd_party
+        requires!(order_data, *SUBMIT_TO_DROP_SHIP_FIREARM_ATTRS[:required])
+
+        endpoint = ENDPOINTS[:submit_to_drop_ship_firearm]
+        order_data = standardize_body_data(order_data, SUBMIT_TO_DROP_SHIP_FIREARM_ATTRS[:permitted])
+      when :drop_ship_to_customer
+        requires!(order_data, *SUBMIT_TO_DROP_SHIP_ATTRS[:required])
+
+        endpoint = ENDPOINTS[:submit_to_drop_ship]
+        order_data = standardize_body_data(order_data, SUBMIT_TO_DROP_SHIP_ATTRS[:permitted])
+      else
+        endpoint = ENDPOINTS[:submit_to_store]
+        order_data = standardize_body_data(order_data, SUBMIT_TO_STORE_ATTRS[:permitted])
+      end
+
```

Comment:
@brownkm5 I would actually leave these broken out as separate methods. The `transfer_to_3rd_party`, `drop_ship_to_customer`, etc. keys are specific to AR. When designing a gem, you want it to be platform agnostic.

Reply:
👌